homeassistant/components/poolstation/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Poolstation integration."""
+from datetime import timedelta
+import logging
+
+import aiohttp
+from pypoolstation import Account, Pool
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import COORDINATORS, DEVICES, DOMAIN, TOKEN
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL = timedelta(seconds=30)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Poolstation from a config entry."""
+    session = async_create_clientsession(hass, cookie_jar=aiohttp.DummyCookieJar)
+    account = Account(session, token=entry.data[TOKEN])
+
+    try:
+        pools = await Pool.all(session, account=account)
+    except aiohttp.ClientError as err:
```

Comment:
I think you should also except `asyncio.TimeoutError` and maybe `aiohttp.ContentTypeError` and if applicable `json.decoder.JSONDecodeError` but not totally sure about this. 

What happens if the token is invalid / expired?

Reply:
@farmio I do not know, I so far my token on the browser hasn't expired since I registered in the service, over a month ago. I don't know if token really expire in this API.

Reply:
`asyncio.TimeoutError` and `aiohttp.ContentTypeError` are causes of `aiohttp.ClientError`.

Reply:
Then this one will catch both. Good to know.

Reply:
This is what I thought too but unfortunately it didn't work as expected. But I was trying it against a local IP address (containing a typo so was not assigned), not a cloud service. See https://github.com/nielstron/pyfronius/pull/36

Reply:
@farmio ok, I added a specific exception handler for `asyncio.TimeoutError`. The error is the same however, as there aren't more specific translations in home assistant for timeouts.